src/driver/tools/mailthread_types.c

```diff
@@ -54,6 +54,11 @@ mailmessage_tree_new(char * node_msgid, time_t node_date,
     return NULL;
 
   tree = malloc(sizeof(* tree));
+  if (tree == NULL) {
```

Comment:
Tks @raniervf ,
The malloc may return NULL if the kernel has no memory to be allocated.